subsys/net/openthread/rpc/server/ot_rpc_netdiag.c

```diff
@@ -310,61 +309,54 @@ static void ot_rpc_cmd_send_diagnostic_get(const struct nrf_rpc_group *group,
 					   struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
 {
 	otIp6Address addr;
-	otError error;
-	uint8_t count;
-	uint8_t tlvTypes[35];
+	otError error = OT_ERROR_INVALID_ARGS;
+	size_t count;
+	const uint8_t *tlvTypes;
 
 	nrf_rpc_decode_buffer(ctx, addr.mFields.m8, OT_IP6_ADDRESS_SIZE);
 
-	count = nrf_rpc_decode_uint(ctx);
+	tlvTypes = nrf_rpc_decode_buffer_ptr_and_size(ctx, &count);
 
-	zcbor_list_start_decode(ctx->zs);
-	for (int i = 0; i < count; i++) {
-		tlvTypes[i] = nrf_rpc_decode_uint(ctx);
+	if (tlvTypes && count && nrf_rpc_decode_valid(ctx)) {
```

Comment:
Is there a reason to check `count`? It seems that the `otThread` functions accept empty TLV arrays and still send some commands in such a case (just without the TLVs).
Also, nit, if `tlvTypes != NULL` implies `nrf_rpc_decode_valid(ctx)` so only checking `tlvTypes` should be enough in this case :).

Reply:
Agree :) changed to checking `tlvTypes` only